server/controllers/documents.js

```diff
@@ -0,0 +1,237 @@
+import { Users, Documents } from '../models';
+import {
+  catchError,
+  returnValidationErrors,
+} from '../helpers/utils';
+
+
+/**
+ * @description creates a document
+ * @function createDocument
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object
+ */
+const createDocument = (req, res) => {
+  req.checkBody('title', 'fullname cannot be empty').notEmpty();
+  req.checkBody('content', 'content cannot be empty').notEmpty();
+  req.checkBody('access', 'access cannot be empty').notEmpty();
+  req.checkBody('access', 'public, private and role are the only allowed acces types')
+    .isIn(['public', 'private', 'role']);
+  req.checkBody('access', 'only letters of the alphabets are allowed as access').isAlpha();
+
+  returnValidationErrors(req, res);
+
+  Documents.create({
+    title: req.body.title,
+    content: req.body.content,
+    access: req.body.access,
+    userId: req.decoded.userId,
+  })
+  .then(
+    document => res.status(201).send({
+      status: 'ok',
+      document,
+      message: 'document was successfully created',
+    })
+  )
```

Comment:
Promises are little bit hard to deal with. the error might happen in as a second arugment to then or in the catch. You should take care of both cases